fedmsg/tests/crypto/test_utils.py

```diff
@@ -105,43 +104,51 @@ def test_topic_missing_not_nitpicky(self):
         self.assertTrue(result)
 
 
-class LoadRemoteCertTests(base.FedmsgTestCase):
+class LoadCertificateTests(base.FedmsgTestCase):
     """Tests for :func:`utils._load_remote_cert`."""
 
     def setUp(self):
-        super(LoadRemoteCertTests, self).setUp()
+        super(LoadCertificateTests, self).setUp()
 
         self.cache_dir = tempfile.mkdtemp()
         self.cache_file = os.path.join(self.cache_dir, 'ca.crt')
         self.addCleanup(shutil.rmtree, self.cache_dir, True)
 
     def test_remote_cert(self):
-        """Assert downloading a certificate to a cache location works."""
+        """Assert requesting a remote certificate works and is cached."""
+        location = 'https://fedoraproject.org/fedmsg/ca.crt'
         with open(os.path.join(base.SSLDIR, 'fedora_ca.crt'), 'r') as fd:
             expected_cert = fd.read()
-        utils._load_remote_cert('https://fedoraproject.org/fedmsg/ca.crt', self.cache_file, 0)
-
-        self.assertTrue(os.path.exists(self.cache_file))
-        with open(self.cache_file, 'r') as fd:
-            actual_cert = fd.read()
-        self.assertEqual(expected_cert, actual_cert)
-
-    @mock.patch('fedmsg.crypto.utils.os.stat')
-    def test_valid_cache(self, mock_stat):
-        """Assert when the primary cache is valid it's used."""
-        mock_stat.return_value.st_mtime = time.time()
-        cache = utils._load_remote_cert('https://example.com/ca.crt', '/my/ca.crt', 60)
-
-        self.assertEqual('/my/ca.crt', cache)
-        mock_stat.assert_called_once_with('/my/ca.crt')
-
-    @mock.patch('fedmsg.crypto.utils.os.stat')
-    def test_valid_alternate_cache(self, mock_stat):
-        """Assert when the alternate cache is valid it's used."""
-        mock_stat.side_effect = [OSError, mock.Mock(st_mtime=time.time())]
-        cache = utils._load_remote_cert('https://example.com/ca.crt', '/my/ca.crt', 60)
-
-        self.assertEqual(os.path.expanduser('~/.local/my/ca.crt'), cache)
-        self.assertEqual('/my/ca.crt', mock_stat.call_args_list[0][0][0])
-        self.assertEqual(
-            os.path.expanduser('~/.local/my/ca.crt'), mock_stat.call_args_list[1][0][0])
+
+        with mock.patch.dict('fedmsg.crypto.utils._cached_certificates', clear=True):
+            ca, crl = utils.load_certificates(location)
+
+            self.assertEqual((expected_cert, None), utils._cached_certificates[location])
+            self.assertEqual(expected_cert, ca)
+            self.assertTrue(crl is None)
```

Comment:
These should also be unindented.

Reply:
I unindented the bottom two, the top one pokes in the mocked dict so it needs to be inside the mock.